Automattic/jetpack
# Search: avoid search query on component mount

Fixes #22231

#### Changes proposed in this Pull Request:
The PR removes search API call on component loaded except for Customberg, and `getResults` still gets called by `onChangeQueryString` on all occasions.

<strike>It also fixed an when overlay is closed, search api is called again.</strike>

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Go to a site with a Search subscription and Instant Search enabled
* Open a frontend page
* Ensure no requests made to `https://public-api.wordpress.com/rest/v1.3/sites/xxx/search`
* Ensure Instant Search load as expected with parameter `?s=hello`
* Ensure Instant Search works
* Ensure Customberg works
